x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/io/stream/PlanNamedTypesTests.java

```diff
@@ -334,15 +331,15 @@ public void testBinComparisonSimple() throws IOException {
         var orig = new Equals(Source.EMPTY, field("foo", DataTypes.DOUBLE), field("bar", DataTypes.DOUBLE));
         BytesStreamOutput bso = new BytesStreamOutput();
         PlanStreamOutput out = new PlanStreamOutput(bso, planNameRegistry);
-        out.writeNamed(BinaryComparison.class, orig);
-        var deser = (Equals) planStreamInput(bso).readNamed(BinaryComparison.class);
+        out.writeNamed(EsqlBinaryComparison.class, orig);
```

Comment:
Does this cause us trouble with the name? I know `PlanNamedTypes` uses the class name sometimes.

Reply:
I think the `PlanNameRegistry` only uses the plain class name, so as long as `Equals extends BinaryComparison` is replaced by `Equals extends EsqlBinaryComparison` - with the same name `Equals`, there's a chance this works without issue.

Changing the "category class" `EsqlBinaryComparison` under which we register `Equals` _could_ in theory break things, but in practice when we read/write expressions, the `PlanNamedTypes` only cares that we have a subclass of `Expression` (which is unchanged).